samples/custom_dispatch/cpu/embedded/example_hal.mlir

```diff
@@ -48,7 +48,8 @@
 #pipeline_layout = #hal.pipeline.layout<push_constants = 1, sets = [
   <0, bindings = [
       <0, storage_buffer, ReadOnly>,
-      <1, storage_buffer>
+      <1, storage_buffer, ReadOnly>,
+      <2, storage_buffer>
```

Comment:
I don't know which commit breaks it, but the fix is reasonable to me. It is required by l.183, which has `binding(2)` while `ordinal 2` not present in descriptor set layout. So the fix is adding one more storage_buffer to the set.

```mlir
          // Bindings are accessed by reference.
          %binding0 = hal.interface.binding.subspan layout(#pipeline_layout) set(0) binding(0) alignment(64) offset(%c0) : memref<?xf32>{%dim}
          %binding1 = hal.interface.binding.subspan layout(#pipeline_layout) set(0) binding(1) alignment(64) offset(%c0) : memref<?xf32>{%dim}
          %binding2 = hal.interface.binding.subspan layout(#pipeline_layout) set(0) binding(2) alignment(64) offset(%c0) : memref<?xf32>{%dim}
```

Reply:
Likely unrelated to this LLVM integrate. I thought I had the ASan job running on all (relevant) PRs and commits, but I had a bug. Fix: https://github.com/iree-org/iree/pull/18108